source/clear-linux/tutorials/dlaas-benchmarks.rst

```diff
@@ -0,0 +1,186 @@
+
+Ensure you have the "containers-basic" bundle installed, this will provide Docker, which is required for TensorFlow benchmarking. Use the :command:`swupd` utility with the `bundle-list` option and check for "containers-basic" in the list:
+
+.. code-block:: bash
+
+  sudo swupd bundle-list
+
+If you need to install the containers-basic bundle, use :command:`swupd` to do so.
+
+.. code-block:: bash
+
+  sudo swupd bundle-add containers-basic
+
+
+To run Kubernetes on |CL|, please refer to the :ref:`kubernetes` tutorial to ensure it is correctly installed and configured.
+
+We have validated these steps against the following software package versions
+
+  * |CL| 26240
+  * Docker 18.06.1
+  * Kubernetes 1.11.3
+  * Go 1.11.12
+
+
+TensorFlow Benchmarks
+=====================
+
+This section describes running the `TensorFlow benchmarks`_ as run by the TensorFlow community. These steps provide a template to run other benchmarks, providing they can invoke TensorFlow.
+
+The |CL| Deep Learning Stack is available in two versions.  First, a version that includes TensorFlow* optimized for Intel Architecture, and second, a version that includes the TensorFlow* framework optimized using Intel® Math Kernel Library for Deep Neural Networks (Intel® MKL-DNN) primitives.
+
+Download and run either the `TensorFlow`_ or the `TensorFlow with MKL-DNN`_  docker image from hub.docker.com. The next commands will take place in the running container. Replace <docker_name> with the name of the image.
+
+First, clone the benchmark repository
+
+.. code-block:: bash
+
+  docker exec -t <docker_name> bash -c ‘git clone http://github.com/tensorflow/benchmarks -b cnn_tf_v1.11_compatible’
+
+Next, execute the benchmark script to run the benchmark
+
+.. code-block:: bash
+
+  docker exec -i <docker_name> bash -c ‘python benchmarks/scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py --device=cpu --model=resnet50 --data_format=NWHC ’.
+
+You can replace the model with one of your choice supported by the TensorFlow benchmarks.
+
+
+
+Kubeflow Benchmarks
+===================
+
+The benchmark workload will run in a Kubernetes (k8s) container. We will use Kubeflow and deploy three nodes for this tutorial, to get a decent return.
+
+k8s setup
++++++++++
+
+Please follow the instructions in the :ref:`kubernetes` tutorial to get set up on |CL|.  The k8s community also has `instructions`_ if you are using Ubuntu instead of |CL|.
+
+
+k8s networking
+++++++++++++++
+
+We have used `flannel`_ as the network provider for these tests. If you are comfortable with another network layer, there is nothing that will prevent you from using that.
+Refer to the Kubernetes `networking documentation`_ for setup.
+
+
+Images
+++++++
+
+We will need to create a docker image that will include launcher.py to include the |CL| Deep Learning Stack, and put the benchmarks repo in the right location. From the docker image, run the following
```

Comment:
We must provide a link to the docker image, it might be confusing just to say "we need to create a docker image". We either provide a link to the Dockerfile and give instructions on how to build the image or provide instructions on how to pull it from docker hub (which I think is easier for the user).  @ganeshmaharaj thoughts?

Reply:
Right, that way is better. That being said, my comment on line 94 is no longer valid. thanks +1